src/Components/WebAssembly/Http/src/Microsoft.AspNetCore.Blazor.HttpClient.csproj

```diff
@@ -12,7 +13,13 @@
   </ItemGroup>
 
   <ItemGroup>
+    <Reference Include="Microsoft.AspNetCore.Components.WebAssembly" />
+    <Reference Include="Microsoft.AspNetCore.Components.WebAssembly.Runtime" PrivateAssets="All" />
     <Reference Include="System.Text.Json" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(ComponentsWebAssemblyFrameworkPath)' != ''">
+    <None Include="$(ComponentsWebAssemblyFrameworkPath)WebAssembly.Net.Http.dll" Pack="true" PackagePath="lib\netstandard2.1" />
```

Comment:
I'm not entirely sure how viable this option is: this change basically surfaces makes types from `WebAssembly.Net.Http.dll` available to consumers of this package at compile time. This assembly is already loaded at runtime since the WASM implementation requires it is present. 

This allows the framework and user code to directly program against the API that .NET WASM provides to configure Fetch Request options. Here's one instance of user code doing this: https://github.com/dotnet/aspnetcore/pull/19253/files#diff-160b59b53a536a446ebfcc5887f97458R55-R65

The nice part of this is that Blazor does not need to ship adapters to these types such as https://github.com/dotnet/aspnetcore/blob/blazor-wasm/src/Components/WebAssembly/WebAssembly/src/Http/FetchCredentialsOption.cs#L9.  The bit I'm unclear about is if this assembly is considered "Private" by .NET Wasm and consequently not meant to be referenced \ subject to change. We should be able to get clarity on this as part of the next sync up.

Note that the Blazor.HttpClient package isn't required for this change. We can put this in the Components.WebAssembly package if we want to.